errors/consts.go

```diff
@@ -31,6 +31,14 @@ const (
 	// Example: the ProviderSpec specifies an instance type that doesn't exist,.
 	InvalidConfigurationMachineError MachineStatusError = "InvalidConfiguration"
 
+	// BootstrapFailedMachineError indicates that the machine has been created, but
+	// the bootstrap process failed (e.g., Cloud init failed),
+	// This is not a transient error, but
+	// indicates a state that must be fixed before progress can be made.
+	//
+	// Example: the ControlPlaneEndpoint is wrong.
+	BootstrapFailedMachineError MachineStatusError = "BootstrapFailed"
```

Comment:
Where would this error be used (on which resource / field / condition) and how do we make sure that users don't confuse this with that the bootstrap provider failed?

Reply:
Ah my bad, confused this with conditions. Just noticed that this would be for the failureReason